### PR TITLE
fix(TDP-4618/TDP-4984): impossible to reopen an existing preparation

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4618
https://jira.talendforge.org/browse/TDP-4984

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/85/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/85/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/86/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/86/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/92/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/92/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/93/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/93/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/94/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/94/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/95/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/95/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/96/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/96/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/97/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/97/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/98/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/98/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/100/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/100/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/101/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/101/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/102/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/102/)

File: dataprep-api/src/test/java/org/talend/dataprep/api/service/TransformAPITest.java [hunk 1]
```diff
@@ -298,8 +299,13 @@ public void testCustomDateFormat_MMMM_yyyy_dd_Transformation() throws Exception
                 .get("/api/preparations/{id}/content?version=head", preparationId)
                 .asString();
 
-        final JsonNode rootNode = mapper.readTree(datasetContent);
-        final DataSetMetadata metadata = mapper.readerFor(DataSetMetadata.class).readValue(rootNode.path("metadata"));
+        final String metadataContent = given().when() //
+                .expect().statusCode(200).log().ifError() //
+                .get("/api/preparations/{id}/metadata?version=head", preparationId)
```
Comment (frassinier) [inline dataprep-api/src/test/java/org/talend/dataprep/api/service/TransformAPITest.java]: comment for style is missing :(

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java [hunk 2]
```diff
@@ -86,13 +87,18 @@ public ExecutableTransformer buildExecutable(DataSet input, Configuration config
...
         final TransformerWriter writer = writerRegistrationService.getWriter(configuration.formatId(), configuration.output(),
                 configuration.getArguments());
+
         final ConfiguredCacheWriter metadataWriter = new ConfiguredCacheWriter(contentCache, DEFAULT);
-        final TransformationMetadataCacheKey metadataKey = cacheKeyGenerator.generateMetadataKey(configuration.getPreparationId(),
+
+        final InitialTransformationMetadataCacheKey metadataKey = cacheKeyGenerator.generateInitialMetadataKey(configuration.getPreparationId(),
                 configuration.stepId(), configuration.getSourceType());
+
+        //TODO: A verifier
```
Comment (frassinier) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: :toilet:

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/PreparationExportStrategy.java [hunk 3]
```diff
@@ -132,6 +135,8 @@ private void performPreparation(final ExportParameters parameters, final OutputS
                             .preparation(preparation) //
                             .stepId(version) //
                             .volume(Configuration.Volume.SMALL) //
+                            .globalStatistics(false)
+                            .allowMetadataChange(true)
```
Comment (frassinier) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/PreparationExportStrategy.java]: comment for style is missing

**Pull request closed without merging**